brokenspoke_analyzer/core/compute.py

```diff
@@ -321,13 +332,20 @@ def conectivity(
             sql_connectivity_script_dir
             / f"reachable_roads_{stress_level}_stress_calc.sql"
         )
-        for i in range(0, 8):
-            bind_params = {
-                "thread_num": 8,
-                "thread_no": i,
-                "nb_max_trip_distance": max_trip_distance,
+        with concurrent.futures.ThreadPoolExecutor() as executor:
+            future_to_road_id = {
+                executor.submit(
+                    execute_sqlfile_with_substitutions,
+                    engine,
+                    sql_script,
+                    {"road_id": i, "nb_max_trip_distance": max_trip_distance},
+                    0.001,
+                ): i
+                for i in road_ids
             }
-            execute_sqlfile_with_substitutions(engine, sql_script, bind_params)
+            for future in concurrent.futures.as_completed(future_to_road_id):
+                url = future_to_road_id[future]
+                data = future.result()
 
```

Comment:
The previous code was doing 8 iterations for calculating the stress level (although I am not sure why 8 was choosen...). Where did this go? We don't need it because we run it once per road id?

Reply:
I wasn't sure on the current implementation. From reading it, it seemed like the modulo 8 meant it was effectively splitting up the entire collection of roads into 8 groups, and each of those groups was updated once. The parameter was named `:thread_num`, but as far as I could tell there were no threads, and it was updating the "groups" sequentially.

Reply:
Ah yes, in the original implementation is was called in parallel. 

You could find it here: https://github.com/PeopleForBikes/pfb-network-connectivity/blob/develop/src/analysis/scripts/run_connectivity.sh#L53-L65